source/faq.txt

```diff
@@ -1,17 +1,103 @@
+.. _node-faq:
+
 ===
 FAQ
 ===
 
-.. default-domain:: mongodb
-
-Frequently Asked Questions
-
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 2
+   :depth: 1
    :class: singlecol
 
+This page contains frequently asked questions and their corresponding answers.
+
+.. tip::
+
+   If you can't find an answer to your problem on this page,
+   see the :ref:`node-issues-help` page for next steps and more
+   resources.
+
+.. _node-faq-connection-pool:
+
+How Does Connection Pooling Work in the Node Driver?
+----------------------------------------------------
+
+Every ``MongoClient`` instance has a built-in connection pool for each server
+in your MongoDB topology. Connection pools open sockets on demand to
+support concurrent requests to MongoDB in your application.
+
+The maximum size of each connection pool is set by the ``maxPoolSize`` option, which
+defaults to ``100``. If the number of in-use connections to a server reaches
+the value of ``maxPoolSize``, the next request to that server will wait
+until a connection becomes available.
+
+Each ``MongoClient`` instance opens two additional sockets per server in your
+MongoDB topology for monitoring the server's state.
+
+For example, a client connected to a 3-node replica set opens 6
+monitoring sockets. It also opens as many sockets as needed to support
+requests on each server, up to the value of ``maxPoolSize``. If
+``maxPoolSize`` is ``100`` and the application only uses the primary
+(the default), then only the primary connection pool grows and there can
+be at most ``106`` total connections. If the application uses a
+:manual:`read preference </core/read-preference/#read-preference>` to
+query the secondary nodes, their pools also grow and there can be
+``306`` total connections.
+
+Additionally, connection pools are rate-limited such that each connection pool
+can only create, at maximum, the value of ``maxConnecting`` connections
+in parallel at any time. Any additional request stops waiting in the
+following cases:
+
+- The connection pool finishes creating a connection and there are fewer
+  than ``minPoolSize`` connections in the pool.
+- An existing connection is checked back into the pool.
+- The driver's ability to reuse existing connections improves due to
+  rate-limits on connection creation.
```

Comment:
[question]

Should this be "Any additional request stops waiting until one of the following cases"? 

Also - for the first case. If the connection pool just finished creating a connection, shouldn't that free up a slot? Because based on the previous wording, you're only talking about exceeding the ``maxConnecting`` parameter. And I think that if the connection pool never exceeded ``maxConnecting`` and just finished a connection, there would be an open spot.

Unless these three options is not meant to refer to exceeding ``maxConnecting``, in which case, they need to be placed elsewhere in the topic.

Reply:
To your first Q-- yes, it reads better with "until". changed the wording.

I'm a little confused at your second question -- my understanding is that there can be at max `maxConnecting` connections CREATED at the same time, and if your app wants to make another connection beyond that number, it will wait until a connection is freed up. 

I do think there is a typo and the first bullet point should read:
- The connection pool finishes creating a connection and there are fewer
  than ``maxPoolSize`` connections in the pool.